client/my-sites/plans/main.jsx

```diff
@@ -39,11 +39,18 @@ var Plans = React.createClass( {
 	},
 
 	componentDidMount: function() {
-		this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
+		this.updateSitePlans();
 	},
 
 	componentWillReceiveProps: function() {
-		this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
+		this.updateSitePlans();
+	},
+
+	updateSitePlans: function() {
+		var selectedSite = this.props.sites.getSelectedSite();
+		if ( selectedSite ) {
```

Comment:
`fetchSitePlans` calls `shouldFetchSitePlans`, which contains this check already.